src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,108 @@
+package duke;
+
+import duke.command.Command;
+import duke.exception.BadDateArgumentException;
+import duke.exception.EmptyArgumentException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDos;
+
+import java.util.List;
+
+public class TaskList {
+    public enum Action{
+        ADD,
+        LIST,
+        DONE,
+        DELETE,
+    }
+    private boolean edited = false;
+    private List<Task> store;
+    public TaskList(List<Task> store) {
+        this.store = store;
+    }
+
+    public List<Task> getRawData(){
+        return this.store;
+    }
+
+    public String run(Command c) throws EmptyArgumentException, BadDateArgumentException {
+        String[] args = c.run();
+        String results;
```

Comment:
Should a more suitable variable name could be used here? Perhaps singular instead of plural?